crates/evm/fuzz/src/strategies/invariants.rs

```diff
@@ -64,27 +68,17 @@ pub fn invariant_strat(
     fuzz_fixtures: FuzzFixtures,
 ) -> impl Strategy<Value = BasicTxDetails> {
     let senders = Rc::new(senders);
-    any::<prop::sample::Selector>()
-        .prop_flat_map(move |selector| {
-            let (contract, func) = {
-                let contracts = contracts.targets.lock();
-                let contracts =
-                    contracts.iter().filter(|(_, (_, abi, _))| !abi.functions.is_empty());
-                let (&contract, (_, abi, functions)) = selector.select(contracts);
-
-                let func = select_random_function(abi, functions);
-                (contract, func)
-            };
-
-            let senders = senders.clone();
-            let fuzz_state = fuzz_state.clone();
-            let fuzz_fixtures = fuzz_fixtures.clone();
-            func.prop_flat_map(move |func| {
-                let sender = select_random_sender(&fuzz_state, senders.clone(), dictionary_weight);
-                let contract =
-                    fuzz_contract_with_calldata(&fuzz_state, &fuzz_fixtures, contract, func);
-                (sender, contract)
-            })
+    any::<prop::sample::Index>()
+        .prop_map(move |index| index.get(&contracts.fuzzed_functions()).clone())
```

Comment:
move `contracts.fuzzed_functions()` outside and merge the two prop* calls

Reply:
> move `contracts.fuzzed_functions()` outside and merge the two prop* calls

makes sense, though I think contracts.fuzzed_functions() should remain inside so new collected contracts / functions to be included?